### PR TITLE
Add Unit Tests for FreeRTOS_DNS_Cache

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Add Unit Tests for FreeRTOS_DNS_Cache

Description
-----------
This change adds the Unit test case for this file along with 100% coverage.

Test Steps
-----------
cmake -S test/unit-test -B test/unit-test/build/ -DCMAKE_BUILD_TYPE=Debug
make -C test/unit-test/build/ all

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c [hunk 1]
```diff
@@ -43,8 +43,7 @@
 #include "mock_FreeRTOS_DNS_Parser.h"
 #include "mock_FreeRTOS_DNS_Networking.h"
 #include "mock_NetworkBufferManagement.h"
-#include "FreeRTOS_DNS.h"
-
+#include "mock_FreeRTOS_DNS.h"
```
Comment (moninom1) [inline test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:46]: This should be #include "FreeRTOS_DNS.h"

Comment (ActoryOu) [reply test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:46]: If that's not the target file to test, we can include mock header instead.

Comment (ActoryOu) [reply test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:46]: But it should incldue FreeRTOS_DNS_Cache.h here.

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c [hunk 2]
```diff
@@ -107,7 +118,7 @@ void test_processDNS_CACHE_CatchAssert( void )
 }
 
 /**
- * @brief Ensures that the same entry is inserted into the cache and retrieved
+ * @brief Ensures that the same entry is inserted into the cache and retrieved .
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:121]:
```suggestion
 * @brief Ensures that the same entry is inserted into the cache and retrieved.
```

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c [hunk 3]
```diff
@@ -339,3 +431,333 @@ void test_processDNS_CACHE_exceed_dns_name_limit( void )
 
     TEST_ASSERT_EQUAL( 0, x );
 }
+
+/**
+ * @brief DNS Lookup success
+ */
+
+
```
[the remark this answers is not in the document]
Comment (ActoryOu) [reply test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:439]: Please remove these two empty lines.

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c [hunk 4]
```diff
@@ -339,3 +431,333 @@ void test_processDNS_CACHE_exceed_dns_name_limit( void )
...
+                         FreeRTOS_htonl( 3 ), pdFALSE, NULL );
+
+    x = Prepare_CacheLookup( "hello", xFamily, ppxAddressInfo );
+    TEST_ASSERT_EQUAL( 0, x );
+}
+
+/**
+ * @brief DNS Lookup success : IPv6
+ */
+
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:620]:
```suggestion
```

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c [hunk 5]
```diff
@@ -339,3 +431,333 @@ void test_processDNS_CACHE_exceed_dns_name_limit( void )
...
+    pxNew_AddrInfo_Stub( xStub_pxNew_AddrInfo );
+
+    x = Prepare_CacheLookup( "hello", xFamily, ppxAddressInfo );
+    TEST_ASSERT_EQUAL( 1, x );
+}
+
+/**
+ * @brief DNS IPv6 Lookup fail : pxAddressInfo = NULL
+ */
+
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:652]:
```suggestion
```

**Review by ActoryOu — APPROVED** [review]

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c (hunk 1 above)
Comment (moninom1) [reply test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:46]: I was under the impression that we have one header for all DNS files, but thats not the case. 
In this case we do not need xStub_pxNew_AddrInfo
We can just use the mocked function.

**Review by moninom1 — APPROVED** [review]

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c [hunk 6]
```diff
@@ -133,6 +146,87 @@ void test_processDNS_CACHE_Success( void )
...
+    pxIP.xIs_IPv6 = pdFALSE;
+
+    xTaskGetTickCount_ExpectAndReturn( 3000 ); /* 3 seconds */
+    FreeRTOS_inet_ntop_ExpectAnyArgsAndReturn( NULL );
+
+    FreeRTOS_dns_update( "hello",
+                         &pxIP,
+                         FreeRTOS_htonl( 3 ), pdFALSE, NULL ); /* lives 3 seconds */
+
+    xTaskGetTickCount_ExpectAndReturn( 10000 );                /* 5 seconds */
```
Comment (tony-josi-aws) [inline test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:195]: The comment and code doesn't seem to match. Isn't it 10 seconds? 
Similarly in line 513, 663, 694.

**Review by tony-josi-aws — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request closed without merging**